src/core/api/index.ts

```diff
@@ -2036,10 +2129,17 @@ class Player extends EventEmitter<PLAYER_EVENT_STRINGS, any> {
   }) : void {
     const { type, period } = value;
 
-    if (type === "audio" || type === "text") {
-      if (this._priv_languageManager) {
-        this._priv_languageManager.removePeriod(type, period);
-      }
+    switch (type) {
+      case "audio" || "text":
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Explicitly specify 2 separate cases that fall through; currently this case clause only works for "audio". ([typescript:S3616](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS3616))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=277&issues=AWUT2m9O8anAXINB3vqm&open=AWUT2m9O8anAXINB3vqm)

[ISSUE_KEY]: AWUT2m9O8anAXINB3vqm